packages/gatsby/src/schema/resolvers.ts

```diff
@@ -566,13 +566,23 @@ export function wrappingResolver<TSource, TArgs>(
       )
       activity.start()
     }
-    try {
-      return resolver(parent, args, context, info)
-    } finally {
+    const result = resolver(parent, args, context, info)
+
+    if (!activity) {
+      return result
+    }
+
+    const endActivity = (): void => {
       if (activity) {
         activity.end()
       }
     }
+    if (typeof result?.then === `function`) {
+      result.then(endActivity, endActivity)
```

Comment:
While this won't swallow the error, I think it will suppress nodejs' builtin reporting for uncaught promises (right? because it sees the error is handled at least somewhere)

Reply:
No, it's a different promise chain than the one returned. So it is not affecting unhandled rejections. Also double-checked with this example in node:

```js
const foo = new Promise(function (resolve, reject) {
  setTimeout(() => {
    reject(new Error("Err"));
  }, 1000);
});

const func = () => {};

foo.then(func, func);

foo.then(() => { console.log(`test`); });
```